app/src/main/java/de/danoeh/antennapod/ui/screen/playback/VariableSpeedDialog.java

```diff
@@ -74,8 +77,28 @@ public void updateSpeed(SpeedChangedEvent event) {
         addCurrentSpeedChip.setText(String.format(Locale.getDefault(), "%1$.2f", event.getNewSpeed()));
     }
 
-    public void updateSkipSilence(boolean skipSilence) {
-        skipSilenceCheckbox.setChecked(skipSilence);
+    public void updateSkipSilence(FeedPreferences.SkipSilence skipSilence) {
+        int id = View.NO_ID;
+        switch (skipSilence) {
+            case OFF: id = R.id.skipSilenceOff; break;
+            case MEDIUM: id = R.id.skipSilenceMedium; break;
+            case AGGRESSIVE: id = R.id.skipSilenceAggressive; break;
+        }
+        skipSilenceToggleGroup.check(id);
+    }
+
+    private void onSkipSilenceChanged(MaterialButtonToggleGroup group, int checkedId, boolean isChecked) {
+        FeedPreferences.SkipSilence opt;
+        final int id = group.getCheckedButtonId();
+        if (id == R.id.skipSilenceOff) opt = FeedPreferences.SkipSilence.OFF;
+        else if (id == R.id.skipSilenceMedium) opt = FeedPreferences.SkipSilence.MEDIUM;
```

Comment:
Please use braces for all `if`/`switch`/etc statements. (I'm a bit surprised that the CI does not complain about this. Will have to check this)